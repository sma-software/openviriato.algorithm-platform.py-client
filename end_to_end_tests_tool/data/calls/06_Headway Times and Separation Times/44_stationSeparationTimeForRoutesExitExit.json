{
    "restResource": "/nodes/{nodeId}/separation-times",
    "httpMethod": "GET",
    "urlParameters": [ "=sepTimeInStation1:trainPathNodes[1].nodeId=" ],
    "queryParameters": [
        {
            "key": "precedingFromNodeTrackId",
            "value": "=sepTimeInStation1:trainPathNodes[1].nodeTrackId=",
            "optional": true
        },
        {
            "key": "precedingToSectionTrackId",
            "value": "=sepTimeInStation1:trainPathNodes[2].sectionTrackId=",
            "optional": true
        },
        {
            "key": "precedingStopStatus",
            "value": "passing",
            "optional": true
        },
        {
            "key": "succeedingFromNodeTrackId",
            "value": "=sepTimeInStation2:trainPathNodes[1].nodeTrackId=",
            "optional": true
        },
        {
            "key": "succeedingToSectionTrackId",
            "value": "=sepTimeInStation2:trainPathNodes[2].sectionTrackId=",
            "optional": true
        },
        {
            "key": "succeedingStopStatus",
            "value": "operationalStop",
            "optional": true
        }
    ],
    "requestBody": null,
    "CSharpMethodName": "GetSeparationTimeInStationForRoutes(precedingTrainRoutingEdge,precedingTrainStopStatus,succeedingTrainRoutingEdge,succeedingTrainStopStatus)",
    "overridenParameterNames": [ "precedingNodeTrackID", "precedingSectionTrackID", "precedingTrainStopStatus", "succeedingNodeTrackID", "succeedingSectionTrackID", "succeedingTrainStopStatus" ],
    "CSharpMethodParameterTypes": [ "SMA.AlgorithmPlatform.AlgorithmInterface.AIDM.IRoutingEdge", "SMA.AlgorithmPlatform.AlgorithmInterface.AIDM.StopStatus", "SMA.AlgorithmPlatform.AlgorithmInterface.AIDM.IRoutingEdge", "SMA.AlgorithmPlatform.AlgorithmInterface.AIDM.StopStatus" ],
    "subcategory": "Get Separation Time in Station for Routes"
}