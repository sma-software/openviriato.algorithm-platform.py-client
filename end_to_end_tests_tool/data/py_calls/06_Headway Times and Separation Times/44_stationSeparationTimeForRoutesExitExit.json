{
  "PythonObjectMapping": [
    {
      "ParameterName": "preceding_train_routing_edge",
      "ClassName": "OutgoingRoutingEdge",
      "PythonParameterMapping": {
        "node_id": "$.urlParameters.[0]",
        "start_node_track_id": "$.queryParameters.[0].value",
        "end_section_track_id": "$.queryParameters.[1].value"
      }
    },
    {
      "ParameterName": "preceding_train_stop_status",
      "ClassName": "convert_string_to_stop_status",
      "PythonParameterMapping": {
        "stop_status_as_string": "$.queryParameters.[2].value"
      }
    },
    {
      "ParameterName": "succeeding_train_routing_edge",
      "ClassName": "OutgoingRoutingEdge",
      "PythonParameterMapping": {
        "node_id": "$.urlParameters.[0]",
        "start_node_track_id": "$.queryParameters.[3].value",
        "end_section_track_id": "$.queryParameters.[4].value"
      }
    },
    {
      "ParameterName": "succeeding_train_stop_status",
      "ClassName": "convert_string_to_stop_status",
      "PythonParameterMapping": {
        "stop_status_as_string": "$.queryParameters.[5].value"
      }
    }
  ]
}