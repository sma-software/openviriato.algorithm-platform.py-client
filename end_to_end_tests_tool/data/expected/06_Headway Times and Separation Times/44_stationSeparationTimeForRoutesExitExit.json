"PT1M30S"